File: 2025_fase2/solutions/sequencia_alex.java
import java.util.*;

public class sequencia_alex {
    static String proximaLinha(String seq) {
        StringBuilder resultado = new StringBuilder();
        int count = 1;

        for (int i = 1; i <= seq.length(); i++) {
            if (i < seq.length() && seq.charAt(i) == seq.charAt(i - 1)) {
                count++;
            } else {
                resultado.append(count).append(seq.charAt(i - 1));
                count = 1;
            }
        }
        return resultado.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        String linha = "1";
        for (int i = 1; i < n; i++) {
            linha = proximaLinha(linha);
        }

        System.out.println(linha);
    }
}

File: 2025_fase2/solutions/sequencia_alex.py
def proxima_linha(seq):
    resultado = []
    cont = 1
    for i in range(1, len(seq)):
        if seq[i] == seq[i - 1]:
            cont += 1
        else:
            resultado.append(str(cont))
            resultado.append(seq[i - 1])
            cont = 1
    # adiciona o último grupo
    resultado.append(str(cont))
    resultado.append(seq[-1])
    return "".join(resultado)

def sequencia_conway(n):
    linha = "1"
    for _ in range(1, n):
        linha = proxima_linha(linha)
    return linha

if __name__ == "__main__":
    n = int(input().strip())  # lê o número da linha
    print(sequencia_conway(n))
